webapp/shop/api/datastore.py

```diff
@@ -0,0 +1,36 @@
+import os
+from typing import Callable, Optional
+
+from google.cloud import datastore
+
+
+def get_datastore_client() -> datastore.Client:
+    return datastore.Client.from_service_account_info(
+        {
+            "token_uri": "https://oauth2.googleapis.com/token",
+            "project_id": os.getenv("GOOGLE_DATASTORE_PROJECT_ID"),
+            "private_key": os.getenv("GOOGLE_DATASTORE_PRIVATE_KEY").replace(
+                "\\n", "\n"
+            ),
+            "client_email": os.getenv("GOOGLE_DATASTORE_EMAIL"),
+        }
+    )
+
+
+def handle_confidentiality_agreement_submission(
+    email: str,
+    client_factory: Callable[[], datastore.Client] = get_datastore_client,
```

Comment:
I would have rather not provided the `client_factory` as a parameter, making the two functions also responsible for creating the `client`.

I would rather pass the `client` directly, even if it makes it a bit annoying to create the `client` in each view.